pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BaseTableUpsertMetadataManager.java

```diff
@@ -55,24 +52,21 @@ public void init(TableConfig tableConfig, Schema schema, TableDataManager tableD
     Preconditions.checkArgument(!CollectionUtils.isEmpty(_primaryKeyColumns),
         "Primary key columns must be configured for upsert enabled table: %s", _tableNameWithType);
 
-    _comparisonColumns = upsertConfig.getComparisonColumns();
-    if (_comparisonColumns == null) {
-      _comparisonColumns = Collections.singletonList(tableConfig.getValidationConfig().getTimeColumnName());
+    _upsertConfig = upsertConfig;
+    List<String> comparisonColumns = upsertConfig.getComparisonColumns();
+    if (comparisonColumns == null) {
+      comparisonColumns = Collections.singletonList(tableConfig.getValidationConfig().getTimeColumnName());
```

Comment:
Is this passed to the partition level metadata manager?

Reply:
good catch. let me add this part.